### ISSUE TITLE
`identifier_name` does not support `allowed_symbols`

### ISSUE THREAD

**Issue by billburgess** [realm/SwiftLint#5271]
### Describe the bug

When using `identifier_name` with `allowed_symbols` you'd expect that the allowed symbol would not throw a warning or error when encountered. This entire category is ignored in the context of `identifier_name` and leaves the user no options at all for allowing even part of this rule. The only option is to keep this rule disabled if you have characters you want to omit from this check. According to the docs this should work. https://realm.github.io/SwiftLint/identifier_name.html

This has been an outstanding issue for at least 5 years and makes this rule completely unusable. 

### Environment
- Version 0.53.0
- Installation via CocoaPods, latest version
```yml
opt_in_rules:
  - identifier_name:
      allowed_symbols:
          - _
```

```swift
// This triggers a violation:
let foo_bar = try! bar()
// error: Identifier Name Violation: Variable name 'foo_bar' should only contain alphanumeric and other allowed characters (identifier_name)
```

**Comment by SimplyDanny** [comment]
Your configuration is incorrect. This is how you need to write it:
```yml
identifier_name:
  allowed_symbols:
    - _
```
